### ISSUE TITLE
Safer raws checks in `declaration-colon-newline-after` rule.

### ISSUE THREAD

**Issue by doberkofler** [stylelint/stylelint#2144]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

After upgrading stylelint from 7.5.0 to 7.6.0 and stylelint-config-standard from 14.0.0 to 15.0.0 I get the following exception when linting files that did not report any errors before

```
Module build failed: TypeError: Cannot read property 'length' of undefined
    at D:\MyDev\ljs_app\trunk\periscope\build\node_modules\stylelint\dist\rules\declaration-colon-newline-after\index.js:25:87
    at D:\MyDev\ljs_app\trunk\periscope\build\node_modules\postcss\lib\container.js:192:28
    at D:\MyDev\ljs_app\trunk\periscope\build\node_modules\postcss\lib\container.js:148:26
    at Rule.each (D:\MyDev\ljs_app\trunk\periscope\build\node_modules\postcss\lib\container.js:114:22)
    at Rule.walk (D:\MyDev\ljs_app\trunk\periscope\build\node_modules\postcss\lib\container.js:147:21)
    at D:\MyDev\ljs_app\trunk\periscope\build\node_modules\postcss\lib\container.js:150:32
    at Root.each (D:\MyDev\ljs_app\trunk\periscope\build\node_modules\postcss\lib\container.js:114:22)
    at Root.walk (D:\MyDev\ljs_app\trunk\periscope\build\node_modules\postcss\lib\container.js:147:21)
    at Root.walkDecls (D:\MyDev\ljs_app\trunk\periscope\build\node_modules\postcss\lib\container.js:190:25)
    at D:\MyDev\ljs_app\trunk\periscope\build\node_modules\stylelint\dist\rules\declaration-colon-newline-after\index.js:19:10
    at D:\MyDev\ljs_app\trunk\periscope\build\node_modules\stylelint\dist\lintSource.js:104:52
```
> Which rule, if any, is this issue related to?

Maybe declaration-colon-newline-after based on the exception ?

> What CSS is needed to reproduce this issue?

```
#back-top {
	position: fixed;
	bottom: 25px;
	margin-left: 0;
	right: 25px;
	text-align: center;
	z-index: 1000;
}

#back-top a {
	display: block;
	text-align: center;
	font: 11px/100% Arial, Helvetica, sans-serif;
	text-transform: uppercase;
	text-decoration: none;
	color: #666;
	transition: background-color 0.3s ease-in;
}

#back-top a:hover {
	color: #000;
}

#back-top span {
	width: 45px;
	height: 45px;
	display: block;
	background: #888 url("../img/up-arrow.png") no-repeat center center;
	border-radius: 5px;
	transition: background-color 0.3s ease-in;
	margin: 0 auto 5px;
}

#back-top a:hover span {
	background-color: #666;
}

#back-top.alignright {
	right: 5px;
}

#back-top.hidden {
	display: none !important;
}

/*media query*/
/* $screen_max = 1095;
$screen_max_em = ($screen_max / 16) * 1em */

@media handheld, only screen and (max-width: 68.4375em) {
	#back-top {
		bottom: 5px;
		width: 45px;
		right: 5px;
	}

	#back-top a:active {
		padding-bottom: 5px;
	}
}
```

> What stylelint configuration is needed to reproduce this issue?

```js
module.exports = {
	'extends': 'stylelint-config-standard',
	'rules': {
		'at-rule-empty-line-before': null,
		'color-hex-case': null,
		'color-hex-length': null,
		'comment-empty-line-before': null,
		'comment-whitespace-inside': null,
		'declaration-block-no-ignored-properties': null,
		'declaration-block-no-shorthand-property-overrides': null,
		'declaration-block-no-duplicate-properties': null,
		'font-family-name-quotes': 'always-where-recommended',
		'function-comma-space-after': null,
		'function-linear-gradient-no-nonstandard-direction': null,
		'function-url-quotes': 'always',
		'indentation': null, // !!! should be set to 'tab'
		'max-empty-lines': null,
		'no-browser-hacks': true,
		'rule-nested-empty-line-before': null, // !!! cannot be used because the proprocessor generates the wrongly formatted code
		'rule-non-nested-empty-line-before': null,
		'selector-combinator-space-after': null,
		'selector-pseudo-element-colon-notation': null,
		'shorthand-property-no-redundant-values': null,
		'string-quotes': 'double',
		'value-list-comma-space-after': null
	}
};
```
> Which version of stylelint are you using?

7.6.0

> How are you running stylelint: CLI, PostCSS plugin, Node API?

postcss-loader in a webpack configuration
```
test: /\.css$/i,
exclude: [/vendor/, /node_modules/],
loader: 'postcss-loader'
```

```
postcss: function () {
	return [autoprefixer, stylelint];
},
```

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

no

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors you are getting)?

exception

**Comment by evilebottnawi** [comment]
@doberkofler `stylus-loader`, we not support officially `stylus`. But you code seem as `scss`, am i right?

**Comment by doberkofler** [comment]
@evilebottnawi I understand that there is no support for stylus but the example is pure css. I actually also wrongly wrote stylus-loader but fixed as with css only the postcss-loader is used in my webpack configuration. The reason why I've case to the conclusion that this might be a problem in stylelint is actually because only by updating stylelint and stylelint-config-standard the exception was fired. When reverting back to the previous version everything runs as expected.

**Comment by evilebottnawi** [comment]
@doberkofler oh sorry, i seem on this lines:
```css
/* $screen_max = 1095;
$screen_max_em = ($screen_max / 16) * 1em */
```
But it is comment :smile:

**Comment by evilebottnawi** [comment]
@doberkofler can not reproduce, can you provide more information or create test repo contain this issue?

**Comment by doberkofler** [comment]
@evilebottnawi This is the smaller example, I was able to come up with. As it is, it should throw the exception and by reverting to stylelint@7.5.0 and stylelint-config-standard@14.0.0, it should work again as expected.

[declaration-colon-newline-after.zip](https://github.com/stylelint/stylelint/files/618949/declaration-colon-newline-after.zip)

**Comment by evilebottnawi** [comment]
@doberkofler Reproduce
Simular problems: https://github.com/stylelint/stylelint/pull/2092. 
We should add check `decl.raws.*` in all rules. Thanks!

**Comment by davidtheclark** [comment]
Closed by #2153.
